src/core/cuda/stream_pool.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <cuda_runtime.h>
+#include "legion.h"
+
+namespace legate {
+namespace cuda {
+
+struct StreamView {
+ public:
+  StreamView(cudaStream_t stream) : valid_(true), stream_(stream) {}
+  ~StreamView();
+
+ public:
+  StreamView(const StreamView&)            = delete;
+  StreamView& operator=(const StreamView&) = delete;
+
+ public:
+  StreamView(StreamView&&);
+  StreamView& operator=(StreamView&&);
```

Comment:
How do you expect we'd use move semantics on a StreamView?

Reply:
People shouldn't move stream views unless they know what they are doing. The intent is to make sure the stream is synchronized only once within the scope.